frontend/class-frontend.php

```diff
@@ -1727,6 +1726,42 @@ protected function get_post_type_archive_title( $separator, $separator_location
 		return $title;
 	}
 
+	/**
+	 * Retrieves the WooCommerce title.
+	 *
+	 * @return string The WooCommerce title.
+	 */
+	protected function get_woocommerce_title() {
+		$post  = get_post( $this->woocommerce_shop_page->get_shop_page_id() );
+		$title = $this->get_seo_title( $post );
+
+		if ( is_string( $title ) && $title !== '' ) {
+			return $title;
+		}
+
+		if ( $this->woocommerce_shop_page->get_shop_page_id() !== -1 && is_archive() ) {
+			$title = $this->get_template( 'title-' . $post->post_type );
+			$title = $this->replace_vars( $title, $post );
+
+			if ( is_string( $title ) && $title !== '' ) {
+				return $title;
+			}
+		}
```

Comment:
Wouldn't be better to just redefine `$title` and use only one check (and return) for `if ( is_string( $title ) && $title !== '' ) { ...` at the end?